### ISSUE TITLE
frontdoor backend-host-header cannot be set to empty

### ISSUE THREAD

**Issue by janabimustafa** [Azure/azure-cli-extensions#2497]
### Extension name (the extension in question)
front-door

### Description of issue (in as much detail as possible)
the parameter --backend-host-header on the commands front-door, front-door backend-pool, front-door backend-pool backend cannot be set to an empty string.

When setting to empty string, the following error is returned: az network front-door backend-pool create: error: argument --backend-host-header: expected one argument.

Per Azure Frontdoor docs, setting the value to an empty string is a valid scenario and it will allow us to utilize a feature of our backend properly.

-----

Thank you!

**Comment by yonzhan** [comment]
front-door

**Comment by cocowalla** [comment]
Having the same issue, and haven't found a workaround, at least not with Azure CLI

**Comment by cocowalla** [comment]
I figured out a work around until this is fixed properly:

```bash
az resource update \
    --name "my-frontdoor" \
    --resource-type "Microsoft.Network/frontdoors" \
    --set "properties.backendPools[0].properties.backends[0].backendHostHeader=''"
```

If you have more backend pools or/and backends, you can run it multiple times, changing the `[0]` index as needed.

**Comment by peterjohnsonme** [comment]
Thank you so much for pointing me in the right direction here.  I used it, but prefer the syntax available to us right on the front-door resource itself instead of the general "az resource update" command (here, parameterized in PowerShell):

```
pool_name="test-pool"
resource_group="test-group"
frontdoor_name="test-frontdoor"
az network front-door update `
	--set BackendPools["name=$pool_name"].backends[0].backendHostHeader='' `
	--resource-group $resource_group `
	--name $frontdoor_name
```

**Comment by cocowalla** [comment]
@peterjohnsonme agreed, that's a better way (docs for `az network front-door update` are practically non-existent, so I hadn't realised it could be used like this) 👍

**Comment by ManishaMNaik** [comment]
Facing the same issue .any solution on this for terraform automation?

**Comment by t-bzhan** [comment]
Fixed with https://github.com/Azure/azure-cli-extensions/pull/3394, the issue should be fixed since 1.0.15. 
The command goes like: 
`az network front-door backend-pool backend update -g FrontDoorExampleResourceGroup -f example-FrontDoor --pool-name exampleBackendBing --index 1 --backend-host-header ""`

Be caution that the backed index should start with 1 and you should use the index returned from the azure CLI command "`az network front-door backend-pool backend list -g FrontDoorExampleResourceGroup -f example-FrontDoor --pool-name exampleBackendBing`":
```
[                                                                 
  {                                                               
    "address": "www.azure.com",                                   
    "backendHostHeader": "www.azure.com",                         
    "enabledState": "Enabled",                                    
    "httpPort": 80,                                               
    "httpsPort": 443,                                             
    "priority": 1,                                                
    "privateEndpointStatus": null,                                
    "privateLinkAlias": null,                                     
    "privateLinkApprovalMessage": null,                           
    "privateLinkLocation": null,                                  
    "privateLinkResourceId": null,                                
    "weight": 50                                                  
  },                                                              
  {                                                               
    "address": "www.bing.com",                                    
    "backendHostHeader": "",                                      
    "enabledState": "Enabled",                                    
    "httpPort": 80,                                               
    "httpsPort": 443,                                             
    "priority": 1,                                                
    "privateEndpointStatus": null,                                
    "privateLinkAlias": null,                                     
    "privateLinkApprovalMessage": null,                           
    "privateLinkLocation": null,                                  
    "privateLinkResourceId": null,                                
    "weight": 50                                                  
  },                                                              
  {                                                               
    "address": "www.baidu.com",                                   
    "backendHostHeader": "www.baidu.com",                         
    "enabledState": "Enabled",                                    
    "httpPort": 80,                                               
    "httpsPort": 443,                                             
    "priority": 1,                                                
    "privateEndpointStatus": null,                                
    "privateLinkAlias": null,                                     
    "privateLinkApprovalMessage": null,                           
    "privateLinkLocation": null,                                  
    "privateLinkResourceId": null,                                
    "weight": 50                                                  
  }                                                               
]          
```                                                       

The backend order in the portal does not respect the order returned from the API and sorted based on the host name for the time being as shown in below image:
![image](https://user-images.githubusercontent.com/61817681/131310139-7b873de1-c976-4602-af9e-2cf3d81cc90d.png)
